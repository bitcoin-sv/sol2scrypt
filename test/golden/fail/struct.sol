// SPDX-License-Identifier: MIT
pragma solidity ^0.8.10;

struct ST {
    uint8 a;
}

struct ST1 {
    uint8 a;
    mapping (address => uint) balances;
}
    
    
contract A {
    struct ST {
        uint8 a;
    }
}

contract B {
    struct ST {
        bool a;
        uint256 x;
    }
}
