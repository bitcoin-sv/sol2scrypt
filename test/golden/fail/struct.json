[
    {
        "src": "test/golden/fail/struct.sol:10:5:10:39",
        "type": "Error",
        "message": "unsupported struct definition with mapping field"
    },
    {
        "src": "test/golden/fail/struct.sol:15:5:17:6",
        "type": "Error",
        "message": "struct `ST` already defined"
    },
    {
        "src": "test/golden/fail/struct.sol:21:5:24:6",
        "type": "Error",
        "message": "struct `ST` already defined"
    }
]